doc/examples/compositing/OSLImageProcessing.gfr

```diff
@@ -0,0 +1,231 @@
+__children["OSLImage1"] = GafferOSL.OSLImage( "OSLImage1" )
+parent.addChild( __children["OSLImage1"] )
+__children["OSLImage1"]["channels"].addChild( Gaffer.NameValuePlug( "", Gaffer.Color3fPlug( "value", defaultValue = imath.Color3f( 1, 1, 1 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ), True, "channel", Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic ) )
+__children["OSLImage1"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ImageReader"] = GafferImage.ImageReader( "ImageReader" )
+parent.addChild( __children["ImageReader"] )
+__children["ImageReader"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["InLayer"] = GafferOSL.OSLShader( "InLayer" )
+parent.addChild( __children["InLayer"] )
+__children["InLayer"].loadShader( "ImageProcessing/InLayer" )
+__children["InLayer"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["PowFloatR"] = GafferOSL.OSLShader( "PowFloatR" )
+parent.addChild( __children["PowFloatR"] )
+__children["PowFloatR"].loadShader( "Maths/PowFloat" )
+__children["PowFloatR"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ColorToFloat"] = GafferOSL.OSLShader( "ColorToFloat" )
+parent.addChild( __children["ColorToFloat"] )
+__children["ColorToFloat"].loadShader( "Conversion/ColorToFloat" )
+__children["ColorToFloat"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["FloatToColor"] = GafferOSL.OSLShader( "FloatToColor" )
+parent.addChild( __children["FloatToColor"] )
+__children["FloatToColor"].loadShader( "Conversion/FloatToColor" )
+__children["FloatToColor"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["PowFloatG"] = GafferOSL.OSLShader( "PowFloatG" )
+parent.addChild( __children["PowFloatG"] )
+__children["PowFloatG"].loadShader( "Maths/PowFloat" )
+__children["PowFloatG"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["PowFloatB"] = GafferOSL.OSLShader( "PowFloatB" )
+parent.addChild( __children["PowFloatB"] )
+__children["PowFloatB"].loadShader( "Maths/PowFloat" )
+__children["PowFloatB"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["Expression1"] = Gaffer.Expression( "Expression1" )
+parent.addChild( __children["Expression1"] )
```

Comment:
In this example would it maybe be worth connecting G/B upto R, rather than a variable in the expression? It feels easier for users to work with the graph as then you can still edit the value in the UI, rather than having to select the expression and manually edit the string?

```
parent["PowFloatG"]["parameters"]["power"] = parent["PowFloatR"]["parameters"]["power"]
parent["PowFloatB"]["parameters"]["power"] = parent["PowFloatR"]["parameters"]["power"]
```

You can also use direct plug connections, but they make the graph look funky when its on a plug thats also exposed in the Graph Editor, in a way you can't recreate by dragging.

Reply:
4b09b29 Superceded by new network.